src/apps/tools/OswAppKcalStats.cpp

```diff
@@ -90,20 +90,20 @@ void OswAppKcalStats::showCurvedChart() {
   hal->gfx()->drawLine(DISP_W / 2, 150 + 15, 120, 220, ui->getPrimaryColor());  
 
   // Data info
+
+  // Show the day of the week that cursor (Dynamic weekDay--info)
+  uint32_t d, wD = 0;
+  hal->getLocalDate(&d, &wD);
+  uint32_t wDay = wD - (6 - this->cursorPos) >= 0 ? wD - (6 - this->cursorPos) : (wD - (6 - this->cursorPos)) + 7;
```

Comment:
I think the graph value can be configured as above code .  Is this unnecessary behavior?

Reply:
`wDay` tells you the day of the week where the current cursor is pointing (calculated as the number of moves from the current day of the week).

```
for (uint8_t Index = 0; Index < 6; Index++) {
    
    x1 = ((DISP_W / 2) - interval * 3) + Index * interval;
    y1 = MIN_VALUE - weekValue[Index];
    x2 = x1 + interval;
    y2 = MIN_VALUE - weekValue[Index + 1];
//...
  }
```

```
for (uint8_t Index = 0; Index < 6; Index++) {
    
    x1 = ((DISP_W / 2) - interval * 3) + Index * interval;
    y1 = MIN_VALUE - calculateKcalorie(getStepsOnDay(wDay));
    x2 = x1 + interval;
    y2 = MIN_VALUE - calculateKcalorie(getStepsOnDay(wDay + 1));
//...
```
We can reconfigure as follows without using the `readyValue()`.

Reply:
When `wd` is 0 and `cursor` are 5, shouldn't `wday` be 6? But if you print it out, you get an unexpected price of `4294967295`.

Reply:
Temporary modification : see (04d9b62bbda02f664607565705e0fe4174cee6a4)